pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -243,5 +249,14 @@ func (p *Provider) getVnetInfoLabels(subnetID string) (map[string]string, error)
 		azureCNIOverlayLabel: strconv.FormatBool(true),
 		podNetworkTypeLabel:  consts.NetworkPluginModeOverlay,
 	}
+
+	// Add stateless CNI label for Kubernetes 1.34+
+	if kubernetesVersion != "" {
+		parsedVersion, err := semver.Parse(kubernetesVersion)
+		if err == nil && parsedVersion.Minor >= 34 {
+			vnetLabels[networkStatelessCNILabel] = "true"
```

Comment:
I went and always created this label setting it to false in < 34
https://github.com/Azure/karpenter-provider-azure/pull/1092/files#r2274701432